src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -144,24 +177,57 @@ public override void TransferFullDuplex(ReadOnlySpan<byte> dataToWrite, Span<byt
             }
         }
 
+        private void TransferWriteOnly(ReadOnlySpan<byte> dataToWrite)
+        {
+            int bitLen = _settings.DataBitLength;
+            int lastBit = bitLen - 1;
+
+            using (StartChipSelect())
+            {
+                for (int i = 0; i < dataToWrite.Length; i++)
+                {
+                    for (int j = 0; j < bitLen; j++)
+                    {
+                        using (StartBitTransfer())
+                        {
+                            int bit = _settings.DataFlow == DataFlow.MsbFirst ? lastBit - j : j;
+                            bool bitToWrite = ((dataToWrite[i] >> bit) & 1) == 1;
+                            WriteBit(bitToWrite);
+                        }
+                    }
+                }
+            }
+        }
+
         private bool ReadWriteBit(bool bitToWrite)
         {
+            // _mosi is checked higher up the call path
             _controller.Write(_mosi, bitToWrite);
             return (bool)_controller.Read(_miso);
         }
 
+        private void WriteBit(bool bitToWrite)
+        {
+            _controller.Write(_mosi, bitToWrite);
+        }
+
         /// <inheritdoc />
         public override void Read(Span<byte> data)
         {
+            if (_miso == -1)
+            {
+                throw new ArgumentException("Cannot read without a miso pin specified");
+            }
+
             Span<byte> dataToWrite = stackalloc byte[data.Length];
             TransferFullDuplex(dataToWrite, data);
         }
 
         /// <inheritdoc />
         public override void Write(ReadOnlySpan<byte> data)
         {
-            Span<byte> dataToRead = stackalloc byte[data.Length];
-            TransferFullDuplex(data, dataToRead);
+            // Output/Write only
```

Comment:
Looking at the function name I don't think it requires any comment (it's probably more confusing than lack of it)